tests/memory_checker/test_memory_checker.py

```diff
@@ -24,59 +32,88 @@
 CONTAINER_STOP_THRESHOLD_SECS = 200
 CONTAINER_RESTART_THRESHOLD_SECS = 180
 CONTAINER_CHECK_INTERVAL_SECS = 1
+WAITING_SYSLOG_MSG_SECS = 130
 
 
-@pytest.fixture(autouse=True)
-def modify_monit_config_and_restart(duthosts, enum_dut_feature_container, enum_rand_one_per_hwsku_frontend_hostname):
+def restart_container(duthost, container_name):
+    """Restarts the container on DuT.
+
+    Args:
+        duthost: The AnsibleHost object of DuT.
+        container_name: A string represents name of the container.
+
+    Returns:
+        None.
+    """
+    logger.info("Restarting '{}' container ...".format(container_name))
+    duthost.shell("systemctl restart {}.service".format(container_name))
+    logger.info("'{}' is restarted.".format(container_name))
 
-    """Backup Monit configuration files, then customize and restart it before testing.
-    Restore original Monit configuration files and restart Monit service after testing.
 
+def backup_monit_config_files(duthost):
+    """Backs up Monit configuration files on DuT.
     Args:
-        duthost: Hostname of DuT.
+        duthost: The AnsibleHost object of DuT.
 
     Returns:
         None.
     """
-    dut_name, container_name = decode_dut_and_container_name(enum_dut_feature_container)
-    pytest_require(dut_name == enum_rand_one_per_hwsku_frontend_hostname and container_name == "telemetry",
-                   "Skips testing memory_checker of container '{}' on the DuT '{}' since another DuT '{}' was chosen."
-                   .format(container_name, dut_name, enum_rand_one_per_hwsku_frontend_hostname))
-    duthost = duthosts[dut_name]
+    logger.info("Backing up Monit configuration files on DuT '{}' ...".format(duthost.hostname))
+    duthost.shell("cp -f /etc/monit/monitrc /tmp/")
+    duthost.shell("mv -f /etc/monit/conf.d/monit_* /tmp/")
+    duthost.shell("cp -f /tmp/monit_telemetry /etc/monit/conf.d/")
```

Comment:
Initially I used `cp -f /etc/monit/conf.d/monit_telemetry /tmp` to only backup Monit configuration file of `telemetry`. However, when I tested this pytest script on a lab device with `20191130.83` image, it failed on two new test cases `test_monit_reset_counter_failure(...)` and `test_monit_new_syntax(...)`. I should answer why it fails with `20191130.83` image?

- Why does this pytest script fail on lab device with `20191130.83` image?
The logic of test function `test_monit_reset_counter_failure(...)` is: **First**, `stress` utility tool is leveraged to consume more than 400MB physical memory in `telemetry` container, then this function will check whether `telemetry` will be restarted or not; **Second**, if `telemetry` container was restarted, `stress` utility is leveraged to consume more than 400MB physical memory to check `telemetry` container will be restarted again or not. **Third**, if `telemetry` container was restarted, this test function will be failed since this restart is not expected. Otherwise, it will succeed in passing the test to prove Monit is unable to reset its internal counter.

  When this test function `test_monit_reset_counter_failure(...)` is executed on lab device with `20191130.83` image, at the **Second** step, the `telemetry` container is restarted. This is unexpected behavior and thus test case is failed at **Third** step.

- Why was `telemetry` container restarted in the **Second** step?
The `telemetry` container is restarted again due to Monit can reset its counter.

- Why Monit can reset its counter in the **Second** step?
In order to answer this question, the prerequisite is to learn `asynchronous` behavior of Monit: https://mmonit.com/monit/documentation/monit.html#PROGRAM%20STATUS%20TEST. One sentence to summarize this `async` behavior is at each polling cycle (every 1 minute in SONiC) Monit will fork a thread to execute the script in configuration file but will collect the running status (exut code) of that script at next polling cycle. For example, Suppose the polling cycle of Monit is 1 minute, it forks a thread to execute the script `memory_checker` at **21:03:00PM**. Then Monit will collect the exit code of this script at **21:04:00PM**.

   At the **First** step, `stress` utility is consuming more than 400MB memory in `telemetry` continuously. `telemetry` is being restarted. At the same time, Monit will run the script `memory_checker`

Reply:
@qiluo-msft  Please also review the above comments and RCA!